c2/c2-client-bundle/c2-client-http/src/main/java/org/apache/nifi/c2/client/http/url/LegacyC2UrlProvider.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.c2.client.http.url;
+
+import static org.apache.commons.lang3.StringUtils.isBlank;
+
+public class LegacyC2UrlProvider implements C2UrlProvider {
+
+    private final String c2Url;
+    private final String c2AckUrl;
+
+    LegacyC2UrlProvider(String c2Url, String c2AckUrl) {
+        this.c2Url = c2Url;
+        this.c2AckUrl = c2AckUrl;
+    }
+
+    @Override
+    public String getHeartbeatUrl() {
+        return c2Url;
+    }
+
+    @Override
+    public String getAcknowledgeUrl() {
+        return c2AckUrl;
+    }
+
+    @Override
+    public String getCallbackUrl(String absoluteUrl, String relativeUrl) throws Exception {
+        if (isBlank(absoluteUrl)) {
+            throw new Exception("Provided absolute url was empty or null. Relative urls are not supported with this configuration");
```

Comment:
Would be better to throw a dedicated exception instead of generic.

Reply:
The exception based solution is replaced with an optional based approach